uxarray/helpers.py

```diff
@@ -248,26 +248,22 @@ def get_all_face_area_from_coords(x,
     -------
     area of all faces : ndarray
     """
-    num_faces = face_nodes.shape[0]
-    area = np.zeros(num_faces)  # set area of each face to 0
-
-    for i in range(num_faces):
-
-        face_z = np.zeros(len(face_nodes[i]))
-
-        face_x = x[face_nodes[i]]
-        face_y = y[face_nodes[i]]
-        # check if z dimension
-        if dim > 2:
-            face_z = z[face_nodes[i]]
 
-        # After getting all the nodes of a face assembled call the  cal. face area routine
-        face_area = calculate_face_area(face_x, face_y, face_z, quadrature_rule,
-                                        order, coords_type)
-
-        area[i] = face_area
-
-    return area
+    areas = xr.apply_ufunc(
```

Comment:
numba can't compile functions with xarray objects so this won't work. 

You'll have to follow https://tutorial.xarray.dev/advanced/apply_ufunc/vectorize_1d.html#high-performance-vectorization-gufuncs-numba-guvectorize and stick the compiled `get_all_face_area_from_coords` in the `apply_ufunc` call

Reply:
Thanks! I can have a look at it later and incorporate. Hi @rajeeja , we can ignore this for now and follow up with other comments in the integration PR #33 to get it merged.